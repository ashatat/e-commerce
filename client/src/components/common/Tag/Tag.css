.tag {
  background: #ea5f50;
  color: white;
  display: inline-block;
  height: 26px;
  line-height: 26px;
  padding: 0 10px 0 28px;
  position: relative;
}

.tag::before {
  content: '';
  background: white;
  border-radius: 50%;
  height: 6px;
  width: 6px;
  position: absolute;
  left: 13px;
  top: 10px;
  box-shadow: inset 0 1px rgba(0, 0, 0, 0.25);
}

.triangle {
  content: '';
  background: white;
  border-bottom: 13px solid transparent;
  border-top: 13px solid transparent;
  border-right: 13px solid #ea5f50;
  position: absolute;
  left: 0;
  top: 0;
}