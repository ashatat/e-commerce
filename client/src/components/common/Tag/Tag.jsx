import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import './Tag.css';

export default function Tag({ name, color, bgColor, className }) {
  return (
    <div className="tag">
      <span
        className={classNames('tag__rectangle', className)}
        style={{ color, backgroundColor: bgColor }}
      >
        {name}
      </span>
      <span className="tag__triangle" style={{ borderRightColor: bgColor }} />
    </div>
  );
}

Tag.propTypes = {
  name: PropTypes.string.isRequired,
  color: PropTypes.string,
  bgColor: PropTypes.string,
  className: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object,
    PropTypes.array,
  ]),
};

Tag.defaultProps = {
  color: null,
  bgColor: null,
  className: null,
};
