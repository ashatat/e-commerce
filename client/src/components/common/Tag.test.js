import React from 'react';
import renderer from 'react-test-renderer';
import Tag from './Tag';

test('Tag componet render', () => {
  const component = renderer
    .create(<Tag name="SALE" color="white" bgColor="#ea5f50" />)
    .toJSON();

  console.log(component);
  expect(component).toMatchSnapshot();
});
