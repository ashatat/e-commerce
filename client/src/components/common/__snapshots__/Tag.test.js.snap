// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag componet render 1`] = `
Array [
  <span
    className="tag"
    style={
      Object {
        "backgroundColor": "#ea5f50",
        "color": "white",
      }
    }
  >
    SALE
  </span>,
  <span
    className="triangle"
    style={
      Object {
        "borderRightColor": "#ea5f50",
      }
    }
  />,
]
`;
